File: src/main/java/by/devpav/serfor/resources/impl/RealmConfigResourceImpl.java
package by.devpav.serfor.resources.impl;

import by.devpav.serfor.domain.dtos.RealmConfigDTO;
import by.devpav.serfor.facade.impl.RealmConfigFacadeImpl;
import by.devpav.serfor.resources.RealmConfigResource;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/realm-configs")
public class RealmConfigResourceImpl extends AbstractBasicEntityResource<RealmConfigDTO>
        implements RealmConfigResource {

    public RealmConfigResourceImpl(RealmConfigFacadeImpl realmConfigFacadeImpl) {
        super(realmConfigFacadeImpl);
    }

}

File: src/main/java/by/devpav/serfor/resources/impl/RealmConfigResourceImpl.kt
package by.devpav.serfor.resources.impl

import by.devpav.serfor.domain.dtos.RealmConfigDTO
import by.devpav.serfor.facade.RealmFacade
import by.devpav.serfor.resources.RealmConfigResource
import by.devpav.serfor.util.HttpResponse
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class RealmConfigResourceImpl : AbstractBasicEntityResource<RealmConfigDTO>(), RealmConfigResource {

    @Autowired
    private lateinit var realmFacade: RealmFacade;


    @GetMapping("/{realmName}/config")
    override fun getRealmConfig(@PathVariable realmName: String): ResponseEntity<RealmConfigDTO> =
            HttpResponse.responseEntity(realmFacade.getRealmConfig(realmName))

}